### ISSUE TITLE
Hunter Pacific RF Fans - Recording Codes in Home Assistant

### ISSUE THREAD

**Issue by AzzaHQ** [smartHomeHub/SmartIR#832]
Hi team,

This is NOT an issue with SmartIR or Home Assistant - I have been successfully using both for Air-Con and media players (using Broadlink RM Mini 3's) for a long time now. Side-note: The SmartIR integration is awesome!! Excellent work to the dev team!! Thank you!!

My issue is with 3 new ceiling fans I was gifted by parents (so had no involvement in decision making).

They are Hunter Pacific "Revolution Attitude" Australia, ceiling fans that I believe operates using RF, and requires each fan to be paired with it's remote.

So upon realising the Broadlink mini can't do RF, I bought a Broadlink RM4 Pro but cannot seem to get it to work to record the fan remote's codes like I have previously done in HA (via recorder script - yes I configured to use RF not default IR). Does anyone know if some home domestic fans do not use the regular 433Mhz RF frequency? Or if there is some trickery that I have to do? Or I need another RF blaster than the Broadlink? Or simply give up as it is likely not possible with my fan?

EDIT: The back of the remote does not really help much. It just says "Hunter Pacific International. IMCP Remote Control".

Any help is most welcome. Sorry if this is the wrong forum to post this question but I am at a loss as to what to do!!

Kind regards,
Aaron

**Comment by cjsmns** [comment]
I've got some Arlec RF fans from Bunnings and had issues learning the codes with the RM4 Pro. I ended up installing a python script on a raspberry pi and learnt the codes that way one at a time. Each time I would use the Broadlink app and tell it to add a new device and then when it came time to learn the code, I would get the app to start scanning, then use the python script on the pi to scan and then press the button on the remote, then do the same as the script needs to confirm the code.

It took some time, but I got there in the end.

Not sure why there was a problem in HA. The learning seems to work fine and I would always use a different command from the remote, but all the codes it learnt did the same thing. Which was odd.

To answer your question, I believe that most fans here in Australia run on the 433.92Mhz frequency. Best thing to do would be to see if you can get the Broadlink app on your phone to see if it can learn the codes, if it does, then try via HA. If the app can't learn the codes, then you are out of luck and need to find another way.

**Comment by AzzaHQ** [comment]
Hello everyone. 

UPDATE: I have since been successful in getting the fans to work. I regret to advise that in this case it was a PEBKAC - where by the problem clearly existed between keyboard and chair! That is, this was my error!!

I did not realise that the RF recorder actually required a long 10 second hold of the button being recorded. Once I figured that out, I managed to successfully record the required buttons for each individually programed fan (both from the Broadlink app and using the Home Assistant SmartIR integration)!

Thank you @cjsmns for you response!

Best wishes everyone!
